.github/workflows/ci.yml

```diff
@@ -53,69 +99,64 @@ jobs:
         with:
           dotnet-version: 5.0.x
       - name: Download Consul
+        if: matrix.component == 'Consul'
         shell: bash
         run: |
           cd Consul.Test
           SYSTEM=$(uname | sed 's/MINGW.*/windows/' | tr A-Z a-z)
-          curl -sSL https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_${SYSTEM}_amd64.zip -o consul.zip
+          curl -sSL https://releases.hashicorp.com/consul/${{ matrix.consul }}/consul_${{ matrix.consul }}_${SYSTEM}_amd64.zip -o consul.zip
           unzip consul.zip
           rm consul.zip
-      - name: Build
-        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+      - name: Build library
```

Comment:
There is no need to build library explicitly, it is enough to build tests (required libraries will be built automatically). If we remove this step we can also remove `libframework`.